examples/shaders/shaders_julia_set.c

```diff
@@ -93,59 +90,63 @@ int main(void)
-            else if (IsKeyPressed(KEY_FOUR)) c[0] = pointsOfInterest[3][0], c[1] = pointsOfInterest[3][1];
-            else if (IsKeyPressed(KEY_FIVE)) c[0] = pointsOfInterest[4][0], c[1] = pointsOfInterest[4][1];
-            else if (IsKeyPressed(KEY_SIX)) c[0] = pointsOfInterest[5][0], c[1] = pointsOfInterest[5][1];
-
-            SetShaderValue(shader, cLoc, c, SHADER_UNIFORM_VEC2);
+            if (IsKeyPressed(k))
+            {
+                // If this key is pressed, set the point of interest of the corresponding key.
+                c[0] = pointsOfInterest[k - KEY_ONE][0];
+                c[1] = pointsOfInterest[k - KEY_ONE][1];
+                SetShaderValue(shader, cLoc, c, SHADER_UNIFORM_VEC2);
+                break;
+            }
         }
 
-        if (IsKeyPressed(KEY_SPACE)) pause = !pause;                 // Pause animation (c change)
-        if (IsKeyPressed(KEY_F1)) showControls = !showControls;  // Toggle whether or not to show controls
-
-        if (!pause)
+        // If "R" is pressed, reset zoom and offset.
+        if (IsKeyPressed(KEY_R))
         {
-            if (IsKeyPressed(KEY_RIGHT)) incrementSpeed++;
-            else if (IsKeyPressed(KEY_LEFT)) incrementSpeed--;
-
-            // TODO: The idea is to zoom and move around with mouse
-            // Probably offset movement should be proportional to zoom level
-            if (IsMouseButtonDown(MOUSE_BUTTON_LEFT) || IsMouseButtonDown(MOUSE_BUTTON_RIGHT))
-            {
-                if (IsMouseButtonDown(MOUSE_BUTTON_LEFT)) zoom += zoom*0.003f;
-                if (IsMouseButtonDown(MOUSE_BUTTON_RIGHT)) zoom -= zoom*0.003f;
-
-                Vector2 mousePos = GetMousePosition();
+            zoom = startingZoom;
+            offset[0] = 0.0;
+            offset[1] = 0.0;
+            SetShaderValue(shader, zoomLoc, &zoom, SHADER_UNIFORM_FLOAT);
+            SetShaderValue(shader, offsetLoc, offset, SHADER_UNIFORM_VEC2);
+        }
 
-                offsetSpeed.x = mousePos.x -(float)screenWidth/2;
-                offsetSpeed.y = mousePos.y -(float)screenHeight/2;
+        if (IsKeyPressed(KEY_SPACE)) incrementSpeed = 0;         // Pause animation (c change)
+        if (IsKeyPressed(KEY_F1)) showControls = !showControls;  // Toggle whether or not to show controls
 
-                // Slowly move camera to targetOffset
-                offset[0] += GetFrameTime()*offsetSpeed.x*0.8f;
-                offset[1] += GetFrameTime()*offsetSpeed.y*0.8f;
-            }
-            else offsetSpeed = (Vector2){ 0.0f, 0.0f };
+        if (IsKeyPressed(KEY_RIGHT)) incrementSpeed++;
+        else if (IsKeyPressed(KEY_LEFT)) incrementSpeed--;
 
+        // If either left or right button is pressed, zoom in/out.
+        const bool leftMouseButtonDown = IsMouseButtonDown(MOUSE_BUTTON_LEFT);
+        if (leftMouseButtonDown ^ IsMouseButtonDown(MOUSE_BUTTON_RIGHT))
```

Comment:
Again, the usage of `^` could be really confusing for a newby, even for me. In raylib I always try to keep the code as much clear and simple as possible. Please, could you review it?

Reply:
I've updated it in [bcc51a2](https://github.com/raysan5/raylib/pull/3467/commits/bcc51a2e7972fe13c0c467ba8140f16023cdd9ac) - is the new version more readable?